core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -121,8 +121,8 @@ public String[] getProbeResultRequirement() {
      * @return folderPath if it valid or return the scm itself
      */
     private String searchPomFiles(Path directory, String pluginName, String scm) {
-        try (Stream<Path> paths = Files.find(directory.resolve("pom.xml"), 2, (path, $) -> path.getFileName().toString().equals("pom.xml"))) {
-            return paths.filter(pom -> pomFileMatchesPlugin(directory, pluginName)).toString();
+        try (Stream<Path> paths = Files.find(directory, 2, (path, $) -> directory.getFileName().toString().equals("pom.xml"))) {
```

Comment:
The `directory` in ` Files.find(directory...` never has a file name. I do need at pass it to `pomFileMatchesPlugin` , I resolve it so that it finds the pom file.

Next I want to return `directory.getFileName()` which will be the `folderPath` that I want to return. 

How do I add it collect and return it.